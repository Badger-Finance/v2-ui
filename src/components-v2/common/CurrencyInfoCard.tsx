import React from 'react';
import { Paper, Typography, makeStyles } from '@material-ui/core';
import Skeleton from '@material-ui/lab/Skeleton';
import BigNumber from 'bignumber.js';
import CurrencyDisplay from './CurrencyDisplay';
import { observer } from 'mobx-react-lite';
import { useContext } from 'react';
import { StoreContext } from 'mobx/store-context';
import { inCurrency } from 'mobx/utils/helpers';

export interface CurrencyInfoCardProps {
	title: string;
	value?: BigNumber;
}

const useStyles = makeStyles((theme) => ({
	infoPaper: {
		padding: theme.spacing(2),
		display: 'flex',
		justifyContent: 'center',
		alignItems: 'center',
		flexDirection: 'column',
	},
}));

const CurrencyInfoCard: React.FC<CurrencyInfoCardProps> = observer((props: CurrencyInfoCardProps) => {
	const classes = useStyles();
	const { title, value } = props;
	const store = useContext(StoreContext);
	const { currency } = store.uiState;
	const displayValue = value ? inCurrency(value, currency) : undefined;
	return (
		<Paper elevation={2} className={classes.infoPaper}>
			<Typography variant="subtitle1" color="textPrimary">
				{title}
			</Typography>
			{displayValue ? (
				<CurrencyDisplay displayValue={displayValue} variant="h5" justifyContent="center" />
			) : (
				<Skeleton animation="wave">
					<Typography variant="h5">Placeholder</Typography>
				</Skeleton>
			)}
		</Paper>
	);
});

export default CurrencyInfoCard;
