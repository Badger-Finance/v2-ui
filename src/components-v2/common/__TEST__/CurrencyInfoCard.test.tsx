import React from 'react';
import CurrencyInfoCard from '../CurrencyInfoCard';
import '@testing-library/jest-dom';
import BigNumber from 'bignumber.js';
import { render } from '@testing-library/react';

describe('CurrencyInfoCard', () => {
	test('Renders correctly', () => {
		const displayPrice = new BigNumber(40);
		const currency = 'usd';
		const { container } = render(
			<CurrencyInfoCard title="Test Price" value={displayPrice} currency={currency} isUsd={true} />,
		);
		expect(container).toMatchSnapshot();
	});
});
