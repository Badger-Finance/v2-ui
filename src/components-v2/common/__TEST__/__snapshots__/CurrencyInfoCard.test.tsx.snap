// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyInfoCard Renders correctly 1`] = `
<div>
  <div
    class="MuiPaper-root makeStyles-infoPaper-1 MuiPaper-elevation2 MuiPaper-rounded"
  >
    <h6
      class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextPrimary"
    >
      Test Price
    </h6>
    <h5
      class="MuiTypography-root MuiTypography-h5"
    >
      $40.00
    </h5>
  </div>
</div>
`;
